### DISCUSSION TITLE
Definition of gaps and displacements in the contact mechanics analyzis

### DISCUSSION THREAD

**Discussion by sannant** [ContactEngineering/topobank#869] [Q&A]
The output of the contact mechanics analyzis needs more documentation. 
Meanwhile I start a discussion here to clarify the questions that come up because of this lack of documentation ? 

Here is what the [README.txt](https://github.com/ContactEngineering/TopoBank/files/9209078/README.txt) to the contactmechanics analysis download currently says about the output quantities: 

- Zero-based index column
- Normalized pressure in units of p/E*
- Fractional contact area in units of A/A0
- Normalized mean gap in units of u/h_rms
- A boolean flag (True/False) which indicates whether the calculation converged
  within the given limit
- Filename of the NetCDF file (order of filenames may be different than index)

So each line also refers to one NetCDF file in the directory, it corresponds to
one external pressure. Inside the NetCDF file you'll find the variables

* `contact_points`: boolean array, true if point is in contact
* `pressure`: floating-point array containing local pressure (in units of `E*`)
* `gap`: floating-point array containing the local gap
* `displacement`: floating-point array containing the local displacements

as well as the attributes

* `mean_pressure`: mean pressure (in units of `E*`)
* `total_contact_area`: total contact area (fractional)

Here is an illustration of the meaning of the variables: 

![](https://user-images.githubusercontent.com/40853209/159238016-c42cf584-7f4a-4b85-a56f-441d84c1a44c.png)

We set the average heights to 0 by detrending, so that the offset is the mean_gap - mean_displacements

Unfortunately, it seems that currently the mean_displacements  is not reported directly. 
So you need either to compute it from the array of displacements in the nc file, or deduce it from the normal force via the Green's function (not sure that's trivial).
 
I opened an issue for that: https://github.com/ContactEngineering/TopoBank/issues/868

Note as well that the offset has only a meaningful physical meaning for non-periodic calculations. 

Related issues: 
https://github.com/ContactEngineering/TopoBank/issues/805
https://github.com/ContactEngineering/TopoBank/issues/281

**Comment by sannant** [comment]
@sitangshugk95 does this answer some of your questions ?

**Reply by sitangshugk95** [reply]
This is very helpful, thank you. I will follow up here with questions (if any).
